sambacc/commands/skips.py

```diff
@@ -0,0 +1,188 @@
+#
+# sambacc: a samba container configuration tool
+# Copyright (C) 2024  John Mulligan
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+
+from typing import Optional
+import argparse
+import os
+
+from sambacc.typelets import Self
+
+from .cli import Context
+
+
+class SkipIf:
+    """Base class for objects used to check if a particular sambacc command
+    should be skipped.
+    Skips are useful when different commands are chained together
+    unconditionally in a configuration file (like k8s init containers) but
+    certain commmands should not be run.
+    """
+
+    NAME: str = ""
+
+    def test(self, ctx: Context) -> Optional[str]:
+        """Return a string explaining the reason for the skip or None
+        indicating no skip is desired.
+        """
+        raise NotImplementedError()  # pragma: nocover
+
+    @classmethod
+    def parse(cls, value: str) -> Self:
+        """Parse a string into a skip class arguments."""
+        raise NotImplementedError()  # pragma: nocover
+
+
+class SkipFile(SkipIf):
+    """Skip execution if a file is, or is not, present. For example,
+    "file:/foo/bar" will skip if the file /foo/bar exists. To skip if a file
+    does not exist, use "file:!/foo/bar" - prefix the file name with an
+    exclaimation point.
+    """
+
+    NAME: str = "file"
+    inverted: bool = False
+    path: str = ""
+
+    @classmethod
+    def parse(cls, value: str) -> Self:
+        obj = cls()
+        if not value:
+            raise ValueError("missing path")
+        if value[0] == "!":
+            obj.inverted = True
+            value = value[1:]
+        obj.path = value
+        return obj
+
+    def test(self, ctx: Context) -> Optional[str]:
+        exists = os.path.exists(self.path)
```

Comment:
Why not use modern `pathlib.Path` ?

Reply:
A mix of force of habit and that the code was already using os.path.